server/app/services/scheduler/filter/affinity.rb

```diff
@@ -152,6 +152,17 @@ def label_match?(node, compare, value)
           false
         end
       end
+
+      # @param [String] val1
+      # @param [String] val2
+      # @return [Boolean]
+      def value_matches?(val1, val2)
```

Comment:
The `val1` and `val2` are not that great here, since they're not symmetric.

What about something more descriptive like `def value_matches?(value, pattern)`?